filestore/s3_test.go

```diff
@@ -30,6 +32,17 @@ type TestSuite struct {
 	deleteTempDir bool
 }
 
+func httpClient() *http.Client {
```

Comment:
Can we have at least one test that uploads / downloads a file where ISV is set to true?

Reply:
We can, but unless we're testing minio with a self-signed cert, the test won't tell us much; I think TLSClientConfig only applies to https URLs.

A better test might be making a self-signed cert and start up minio with it.  In that scenario we'd want to default InsecureSkipVerify to true in s3_test.go, and do one test with it set to false and make sure it fails.  That's a bit more work but I think it's worth it.